packages/kbn-legacy-logging/src/legacy_logging_server.ts

```diff
@@ -84,22 +95,19 @@ export class LegacyLoggingServer {
 
   private onPostStopCallback?: () => void;
 
-  constructor(legacyLoggingConfig: Readonly<LegacyVars>) {
+  constructor(legacyLoggingConfig: any) {
     // We set `ops.interval` to max allowed number and `ops` filter to value
     // that doesn't exist to avoid logging of ops at all, if turned on it will be
     // logged by the "legacy" Kibana.
-    const config = {
-      logging: {
-        ...legacyLoggingConfig,
-        events: {
-          ...legacyLoggingConfig.events,
-          ops: '__no-ops__',
-        },
+    const { value: loggingConfig } = legacyLoggingConfigSchema.validate({
+      ...legacyLoggingConfig,
+      events: {
+        ...legacyLoggingConfig.events,
+        ops: '__no-ops__',
       },
-      ops: { interval: 2147483647 },
-    };
+    });
 
-    setupLogging(this, Config.withDefaultSchema(config));
+    setupLogging((this as unknown) as Server, loggingConfig, 2147483647);
```

Comment:
So we expect `LegacyLoggingServer` won't be called when the package is imported from CLI scripts? How does the final dependency tree look like?

Reply:
The only import of this package in core (soon in `kbn/config` instead) is from the legacy appender

https://github.com/elastic/kibana/blob/ab92bbb726b806f4f0c1daa6320baf61fbb1bf96/src/core/server/legacy/logging/appenders/legacy_appender.ts#L48-L50

The idea was to change the file to use a dynamic `require` in the constructor to only load the module when needed. The cli / packages that will use the config service would default to KP logging (probably just defaulting to console appender), that way, the legacy logging (and hapi stuff) is not imported.

Note that the main goal to not load the legacy logging system / hapi when importing the config service was for the apm script instrumentation, which is no longer concerned by the problem, as it will have its own loader to ensure we are not importing any module we want to instrument at all (https://github.com/elastic/kibana/pull/77855)